src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/SourceMappingParser.cs

```diff
@@ -55,28 +56,22 @@ public async Task<IReadOnlyCollection<SourceMapping>> ParseMappings()
         var settings = await JsonSerializer.DeserializeAsync<SourceMappingFile>(stream, options)
             ?? throw new Exception($"Failed to deserialize {VmrInfo.SourceMappingsFileName}");
 
-        var patchesPath = settings.PatchesPath;
-        if (patchesPath is not null)
-        {
-            if (patchesPath.Contains('\\') || patchesPath.StartsWith('/'))
-            {
-                throw new Exception($"Invalid value '{patchesPath}' for {VmrInfo.SourceMappingsFileName} attribute {nameof(settings.PatchesPath)}! " +
-                    $"The path must be relative to the VMR directory and use UNIX directory separators (e.g. src/installer/patches).");
-            }
-
-            _vmrInfo.PatchesPath = Path.Combine(_vmrInfo.VmrPath, patchesPath.Replace('/', Path.DirectorySeparatorChar));
-        }
+        _vmrInfo.PatchesPath = NormalizePath(settings.PatchesPath);
 
-        var contentPath = settings.ContentPath;
-        if (contentPath is not null)
+        if (settings.AdditionalMappings is not null)
         {
-            if (contentPath.Contains('\\') || contentPath.StartsWith('/'))
+            var additionalMappings = new List<(string Source, string? Destination)>();
+            foreach (var additionalMapping in settings.AdditionalMappings)
             {
-                throw new Exception($"Invalid value '{contentPath}' for {VmrInfo.SourceMappingsFileName} attribute {nameof(settings.ContentPath)}! " +
-                    $"The path must be relative to the VMR directory and use UNIX directory separators (e.g. src/installer/content).");
+                if (additionalMapping.Source is null)
+                {
+                    throw new Exception($"Invalid additional mapping: {additionalMapping.Source} -> {additionalMapping.Destination}");
```

Comment:
Would it not be clearer to just specify that the source of the mapping is missing?

Reply:
Right. I could probably improve this validation even further.